src/screens/Login/index.tsx

```diff
@@ -1,15 +1,123 @@
+      }
+    }
+  };
+  const handleRegister = () => {
+    navigation.navigate('Register'); // Replace 'Register' with the name of your registration screen or route
+  };
+  const isFormValid = !errors.email && !errors.password; // Check if both email and password fields have no errors
 
-function Login() {
   return (
-    // <Column safeAreaTop mt={30} h="100%">
-    <Center safeAreaTop mt={30} h="100%">
-      <Heading textAlign="center" mb={6}>
-        Welcome to CraftPile!
-      </Heading>
-    </Center>
-    // </Column>
+      <Center flex={1} py="10">
+        <Box w="90%">
+          <Heading size="lg" fontWeight="600" color="coolGray.800" _dark={{ color: 'warmGray.50' }}>
+            Welcome to CraftPile!
+          </Heading>
+          <Heading mt="1" _dark={{ color: 'warmGray.200' }} color="coolGray.600" fontWeight="medium" size="xs">
+            Sign in to continue!
+          </Heading>
+          <VStack space={3} mt="5">
+            <FormControl isInvalid={!!errors.email?.message}>
+              <FormControl.Label>Email ID</FormControl.Label>
+              <Controller
+                control={control}
+                name="email"
+                render={({ field: { onChange, onBlur, value } }) => (
+                  <Input autoCapitalize="none" onChangeText={onChange} onBlur={onBlur} value={value} />
+                )}
+                rules={{
+                  required: { value: true, message: 'Email is required' },
+                  pattern: { value: emailPattern, message: 'Invalid email address' },
+                }}
+              />
+              {errors.email && <FormControl.ErrorMessage>{errors.email.message}</FormControl.ErrorMessage>}
+            </FormControl>
+            <FormControl isInvalid={!!errors.password?.message}>
+              <FormControl.Label>Password</FormControl.Label>
+              <Controller
+                control={control}
+                name="password"
+                render={({ field: { onChange, onBlur, value } }) => (
+                  <Input onChangeText={onChange} onBlur={onBlur} value={value} secureTextEntry />
+                )}
+                rules={{
+                  required: { value: true, message: 'Password is required' },
+                  pattern: {
+                    value: passwordPattern,
+                    message: 'Password must contain at least 8 characters, including at least one uppercase letter, one lowercase letter, and one number',
+                  },
+                }}
+              />
+              <Link _text={{ fontSize: 'xs', fontWeight: '500', color: 'indigo.500' }} alignSelf="flex-end" mt="1">
+                Forget Password?
+              </Link>
+              {errors.password && <FormControl.ErrorMessage>{errors.password.message}</FormControl.ErrorMessage>}
+            </FormControl>
+            {errorMessage ? <Text color="red.500">{errorMessage}</Text> : null}
```

Comment:
Native base provides an error text component. Will that work instead of the `Text color red 500`?
```js
<FormControl.ErrorMessage _text={{
        fontSize: 'xs'
      }}>
```